drivers/lora/sx1276.c

```diff
@@ -296,47 +309,38 @@ void SX1276SetRfTxPower(int8_t power)
 		return;
 	}
 
-	pa_config = (pa_config & RF_PACONFIG_MAX_POWER_MASK) | 0x70;
+	pa_config &= RF_PACONFIG_MAX_POWER_MASK;
+	pa_config &= RF_PACONFIG_OUTPUTPOWER_MASK;
 	pa_config &= RF_PACONFIG_PASELECT_MASK;
 
+	pa_dac &= RF_PADAC_20DBM_MASK;
+
 #if defined CONFIG_PA_BOOST_PIN
-	pa_config |= RF_PACONFIG_PASELECT_PABOOST;
+	power = clamp_s8(power, 2, 20);
 
+	pa_config |= RF_PACONFIG_PASELECT_PABOOST;
 	if (power > 17) {
-		pa_dac = (pa_dac & RF_PADAC_20DBM_MASK) | RF_PADAC_20DBM_ON;
+		pa_dac |= RF_PADAC_20DBM_ON;
+		pa_config |= (power - 5) & 0x0F;
 	} else {
-		pa_dac = (pa_dac & RF_PADAC_20DBM_MASK) | RF_PADAC_20DBM_OFF;
+		pa_dac |= RF_PADAC_20DBM_OFF;
+		pa_config |= (power - 2) & 0x0F;
 	}
+#elif CONFIG_PA_RFO_PIN
+	power = clamp_s8(power, -4, 15);
 
-	if ((pa_dac & RF_PADAC_20DBM_ON) == RF_PADAC_20DBM_ON) {
-		if (power < 5) {
-			power = 5;
-		} else if (power > 20) {
-			power = 20;
-		}
-
-		pa_config = (pa_config & RF_PACONFIG_OUTPUTPOWER_MASK) |
-			     ((power - 5) & 0x0F);
+	pa_dac |= RF_PADAC_20DBM_OFF;
+	if (power > 0) {
+		/* Set the power range to 0 -- 10.8+0.6*7 dBm. */
+		pa_config |= 7 << SX1276_PA_CONFIG_MAX_POWER_SHIFT;
+		pa_config |= (power & 0x0F);
 	} else {
-		if (power < 2) {
-			power = 2;
-		} else if (power > 17) {
-			power = 17;
-		}
-
-		pa_config = (pa_config & RF_PACONFIG_OUTPUTPOWER_MASK) |
-			     ((power - 2) & 0x0F);
+		/* Set the power range to -4.2 -- 10.8+0.6*0 dBm */
+		pa_config |= 0 << SX1276_PA_CONFIG_MAX_POWER_SHIFT;
```

Comment:
this is a nop, please remove.

Reply:
We probably want to keep the comment in that case since the MAX_POWER field affects how the power is calculated. The reason I added the line in the first place was as documentation.

Reply:
Yes, let's keep the comment.